### PR TITLE
Typos: fix typos in code

### PR DESCRIPTION
**Pull request by mykaul**
Fixes some more typos as found by codespell run on the code. In this commit, there are more user-visible errors.

Refs: https://github.com/scylladb/scylladb/issues/16255

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -1984,7 +1984,7 @@ int main(int ac, char** av) {
         return 1;
     }
 
-    // Even on the environment which causes error during initalize Scylla,
+    // Even on the environment which causes error during initialize Scylla,
```
Comment (mykaul) [inline main.cc:1987]:
```suggestion
    // Even on the environment which causes errors during Scylla initialization,
```

File: unified/build_unified.sh [hunk 2]
```diff
@@ -80,7 +80,7 @@ for pkg in $PKGS; do
     dirname=$(basename "$pkg"| sed -e "s/-$VERSION_ESC-$RELEASE_ESC\.[^.]*\.tar\.gz//")
     dirname=${dirname/#$PRODUCT/scylla}
     if [ ! -d "$BASEDIR/$dirname" ]; then
-        echo "Directory $dirname not found in $pkg, the pacakge may corrupted."
+        echo "Directory $dirname not found in $pkg, the package may corrupted."
```
Comment (mykaul) [inline unified/build_unified.sh:83]:
```suggestion
        echo "Directory $dirname not found in $pkg, the package might be corrupted."
```

File: cql3/statements/alter_table_statement.cc [hunk 3]
```diff
@@ -332,7 +332,7 @@ std::pair<schema_builder, std::vector<view_ptr>> alter_table_statement::prepare_
                 const auto& cdc_opts = dynamic_pointer_cast<cdc::cdc_extension>(it->second)->get_options();
                 if (!cdc_opts.is_enabled_set()) {
                     // "enabled" flag not specified
-                    throw exceptions::invalid_request_exception("Altering CDC options requires specyfing \"enabled\" flag");
+                    throw exceptions::invalid_request_exception("Altering CDC options requires specifying \"enabled\" flag");
```
Comment (nyh) [inline cql3/statements/alter_table_statement.cc:335]: These changes error messages might trip test/cql and dtest tests which often check exact error messages, but I guess we'll just see if anything fails, or not.

**Review by nyh — APPROVED** [review]

**Comment by nyh** [comment]
Looks good. I want to wait for the CI because I'm worried we may have tests that rely on the error messages with the spelling mistakes in them. I don't see CI running yet, hopefully it will start soon?

File: cql3/statements/alter_table_statement.cc (hunk 3 above)
Comment (mykaul) [reply cql3/statements/alter_table_statement.cc:335]: Yes, I was concerned exactly about this, which is why I've split the PRs (not to mention hundreds of changes in a single PR are a nightmare to review).

**Pull request merged**